privatelink-with-network-firewall.yaml

```
@@ -0,0 +1,746 @@
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId:
+        Ref: rEgressVpc
+      CidrBlock:
+        Ref: pEgressVpcPrivateSubnetCidrBlock
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-private-subnet"]]
+  # Public subnet in the egress VPC
+  rEgressVpcPublicSubnet:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId:
+        Ref: rEgressVpc
+      CidrBlock:
+        Ref: pEgressVpcPublicSubnetACidrBlock
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-public-subnet"]]
+  # Internet gateway
+  rInternetGateway:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-igw"]]
+  rInternetGatewayEgressAttachment:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId:
+        Ref: rEgressVpc
+      InternetGatewayId:
+        Ref: rInternetGateway
+  # NAT gateway
+  rElasticIP:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-public-nat-eip"]]
+  rNATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId:
+        Fn::GetAtt:
+          - rElasticIP
+          - AllocationId
+      SubnetId:
+        Ref: rEgressVpcPublicSubnet
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pRosaClusterName, "-egress-public-nat"]]
+
+  # Network Firewall
+  rEgressFirewall:
+    Type: AWS::NetworkFirewall::Firewall
+    Properties:
+      FirewallName: !Join ["", [!Ref pRosaClusterName, "-network-firewall"]]
+      VpcId: !Ref rEgressVpc
+      SubnetMappings:
+        - SubnetId: !Ref rEgressVpcPrivateSubnet
+      DeleteProtection: false
+      FirewallPolicyChangeProtection: false
+      SubnetChangeProtection: false
+      Tags:
+        - Key: "Name"
+          Value: !Join ["", [!Ref pRosaClusterName, "-network-firewall"]]
+      Description: Network Firewall for Egress VPC
+      FirewallPolicyArn: !Ref rEgressFirewallPolicy
+
+  rEgressFirewallPolicy:
+    Type: AWS::NetworkFirewall::FirewallPolicy
+    Properties:
+      FirewallPolicyName:
+        !Join ["", [!Ref pRosaClusterName, "-firewall-policy"]]
+      FirewallPolicy:
+        StatelessDefaultActions:
+          - "aws:forward_to_sfe"
+        StatelessFragmentDefaultActions:
+          - "aws:forward_to_sfe"
+        StatefulRuleGroupReferences:
+          - ResourceArn: !Ref rEgressFirewallRuleGroup
+      Description: Stateful rule group for Network Firewall
+
+  rEgressFirewallRuleGroup:
+    Type: "AWS::NetworkFirewall::RuleGroup"
+    Properties:
+      Capacity: "100"
+      RuleGroupName: !Join ["", [!Ref pRosaClusterName, "-firewall-rule-group"]]
+      Type: "STATEFUL"
+      Tags:
+        - Key: "Name"
+          Value: !Join ["", [!Ref pRosaClusterName, "-firewall-rule-group"]]
+      RuleGroup:
```

I think you also need to set the `HOME_NET` variable to filter traffic from the ROSA VPC.
See https://docs.aws.amazon.com/network-firewall/latest/developerguide/stateful-rule-groups-domain-names.html#stateful-rule-groups-domain-names-home-net

```suggestion
      RuleGroup:
        RuleVariables:
          IPSets:
            HOME_NET:
              Definition:
                - "10.0.0.0/16"
                - "10.1.0.0/16"
```

---

Thanks, added